### ISSUE TITLE
Path append is removing leading slash

### ISSUE THREAD

**Issue by Deliganli** [fs2-blobstore/fs2-blobstore#959]
Hi!

Thanks for the great library, I am playing around a little bit, sorry if it is misuse.

Noticed that if a absolute path is added to `file://` url, the leading slash is removed.

In my case I was parsing the absolute path (`/my/dir/test/`) from config, and appending to the url, and it wasn't finding anything.

```scala
import blobstore.fs.FileStore
import blobstore.url.Url
import cats.effect.IO
import cats.implicits._
import cats.effect.unsafe.implicits.global

val suffix = "/my/dir/test/"
// suffix: String = "/my/dir/test/"
val a      = Url.unsafe("file://") / suffix
// a: Url.Plain = Url(
//   scheme = "file",
//   authority = Authority(
//     host = Hostname(labels = Singleton(a = Label(value = "localhost"))),
//     userInfo = None,
//     port = None
//   ),
//   path = RootlessPath(
//     representation = "my/dir/test/",
//     segments = Append(leftNE = Wrap(seq = List("my", "dir", "test")), rightNE = Singleton(a = ""))
//   )
// )
val b      = Url.unsafe(s"file://$suffix")
// b: Url.Plain = Url(
//   scheme = "file",
//   authority = Authority(
//     host = Hostname(labels = Singleton(a = Label(value = "localhost"))),
//     userInfo = None,
//     port = None
//   ),
//   path = AbsolutePath(
//     representation = "/my/dir/test/",
//     segments = Wrap(seq = List("my", "dir", "test", ""))
//   )
// )

a == b
// res0: Boolean = false

val fs = FileStore[IO].lift(_.path.valid)
// fs: blobstore.Store[IO, blobstore.fs.NioPath] = blobstore.Store$DelegatingStore@26024048

fs.list(a).compile.toList.unsafeRunSync()
// res1: List[Url[blobstore.fs.NioPath]] = List()
fs.list(b).compile.toList.unsafeRunSync()
// res2: List[Url[blobstore.fs.NioPath]] = List(
//   Url(
//     scheme = "file",
//     authority = Authority(
//       host = Hostname(labels = Singleton(a = Label(value = "localhost"))),
//       userInfo = None,
//       port = None
//     ),
//     path = AbsolutePath(
//       representation = NioPath(
//         path = /my/dir/test/test1.txt,
//         size = Some(value = 0L),
//         isDir = false,
//         lastModified = Some(value = 2023-09-19T08:25:14.874880710Z)
//       ),
//       segments = Wrap(seq = List("my", "dir", "test", "test1.txt"))
//     )
//   )
// )
```

Anyways not sure if it is intended, thanks again

**Comment by gafiatulin** [comment]
Hi @Deliganli! 
I think original problem is with the `file://`. According to https://datatracker.ietf.org/doc/html/rfc3986#section-3.3 it should be unrepresentative since it is ambiguous: it might be either a valid URL with a zero-length authority and a zero-length path, or an invalid URI that omits the authority and has a path that starts with two forward slashes.

So correct thing here would be to fail parsing `file://` (it already happens for other schemas, but not for file because it is a bit of a special case).

**Comment by gafiatulin** [comment]
Change has been released as [v0.8.11](https://github.com/fs2-blobstore/fs2-blobstore/releases/tag/v0.8.11) and [v0.9.12](https://github.com/fs2-blobstore/fs2-blobstore/releases/tag/v0.9.12)
